core/embed/rust/src/ui/macros.rs

```diff
@@ -4,3 +4,16 @@ macro_rules! include_res {
         include_bytes!(concat!(env!("CARGO_MANIFEST_DIR"), "/src/ui/", $filename))
     };
 }
+
+/// Concatenates slices and prints them into a terminal.
+#[allow(unused_macros)] // Should be used only for debugging purposes
+macro_rules! print {
```

Comment:
...then we can still use this for convenience, but it can just convert to a sequence of `print()` calls instead of constructing a string.

for bonus points, `println!` can be `print!(...); print("\n")` :)